src/test/java/jenkins/plugins/logstash/LogstashNotifierTest.java

```diff
@@ -124,36 +94,20 @@ public void performSuccess() throws Exception {
     // Verify results
     assertTrue("Build should not be marked as failure", result);
 
-    verify(mockBuild).getId();
-    verify(mockBuild, times(2)).getResult();
-    verify(mockBuild, times(2)).getParent();
-    verify(mockBuild, times(2)).getDisplayName();
-    verify(mockBuild).getFullDisplayName();
-    verify(mockBuild).getDescription();
-    verify(mockBuild).getUrl();
-    verify(mockBuild).getAction(AbstractTestResultAction.class);
-    verify(mockBuild).getBuiltOn();
-    verify(mockBuild, times(2)).getNumber();
-    verify(mockBuild).getTimestamp();
-    verify(mockBuild, times(3)).getRootBuild();
-    verify(mockBuild).getBuildVariables();
-    verify(mockBuild).getLog(3);
-
-    verify(mockTestResultAction).getTotalCount();
-    verify(mockTestResultAction).getSkipCount();
-    verify(mockTestResultAction).getFailCount();
-    verify(mockTestResultAction, times(2)).getFailedTests();
-
-    verify(mockProject, times(2)).getName();
-
-    verify(mockDao).buildPayload(Matchers.any(BuildData.class), Matchers.eq("http://my-jenkins-url"), Matchers.anyListOf(String.class));
-    verify(mockDao).push("{}");
```

Comment:
Excellent! I cringed when I wrote these tests and almost refactored the code again to get better encapsulate the logic. Now it's done.